packages/@aws-cdk/aws-codedeploy/lib/deployment-group.ts

```diff
@@ -99,6 +99,41 @@ class ImportedServerDeploymentGroupRef extends ServerDeploymentGroupRef {
   }
 }
 
+/**
+ * Represents a group of instance tags.
+ * An instance will match a group if it has a tag matching
+ * any of the group's tags by key and any of the provided values -
+ * in other words, tag groups follow 'or' semantics.
+ * If the value for a given key is an empty array,
+ * an instance will match when it has a tag with the given key,
+ * regardless of the value.
+ * If the key is an empty string, any tag,
+ * regardless of its key, with any of the given values, will match.
+ */
+export type InstanceTagGroup = {[key: string]: string[]};
+
+/**
+ * Represents a set of instance tag groups.
+ * An instance will match a set if it matches all of the groups in the set -
+ * in other words, sets follow 'and' semantics.
+ * You can have a maximum of 3 tag groups inside a set.
+ */
+export class InstanceTagSet {
```

Comment:
Why is this type at all needed? Makes usage more convoluted without adding much value. I would expect:

```ts
new DeploymentGroup(this, 'gd', {
  ec2InstanceTags: {
    'key1': [ 'tag1', 'tag2' ]
  }
});
```

Also, since this is an array, we have seen it's useful to have a mutator, so add a method `addEc2InstanceTag(tag, ...groups)` and `addOnPermiseInstanceTag(tag, ...group)`.

Reply:
Because you need to have 'and' and 'or' semantics represented. Have you seen the example from the ReadMe I gave?

```ts
    onPremiseInstanceTags: new codedeploy.InstanceTagSet(
        // only instances with tags (key1=v1 or key1=v2) AND key2=v3 will match this set
        {
            'key1': ['v1', 'v2'],
        },
        {
            'key2': ['v3'],
        },
    ),
```